src/Control/Foldl.hs

```diff
@@ -1048,6 +1049,17 @@ handles k (Fold step begin done) = Fold step' begin done
     step' = flip (appEndo . getDual . getConst . k (Const . Dual . Endo . flip step))
 {-# INLINABLE handles #-}
 
+{- | @{foldOfWith f folder xs} folds all values from a Lens, Traversal, Prism or Fold with the given folder
+
+> foldOfWith f folder xs == L.fold folder (xs^..f)
+
+> foldOfWith (folded.f) folder xs == L.fold (handles f folder) xs
+
+-}
+foldOfWith :: (forall f. (Contravariant f, Applicative f) => (a -> f a) -> s -> f s) -> Fold a b -> s -> b
+foldOfWith l (Fold step begin done) =
+  done . flip appEndo begin . getDual . getConst . l (Const . Dual . Endo . flip step)
```

Comment:
my test code had it's own foldOfWith defined and looking at ghc-core's output it got inlined everywhere (it's just a bunch of newtype wrappers and some simple function calls)